[[ Bug 22017 ]] Fix memory leak when updating listBehavior fields

runrevmark opened the pull request:
This patch fixes a memory leak which can occur when updating listBehavior
fields in some cases. The leak was caused by failing to release the saved
set of previously hilited lines after re-applying them. It has been fixed
by passing a reference to an appropriate auto-class around, rather than
bare pointers.

livecodeali commented:
@livecode-vulcan review ok 7f87f62

livecodeali commented:
This looks fine, just wondering if it's worth adding tests for https://quality.livecode.com/show_bug.cgi?id=2381 and https://quality.livecode.com/show_bug.cgi?id=2766 at the same time seeing as it touches that code

runrevmark committed:
[[ Bug 22017 ]] Fix memory leak when updating listBehavior fields

This patch fixes a memory leak which can occur when updating listBehavior
fields in some cases. The leak was caused by failing to release the saved
set of previously hilited lines after re-applying them. It has been fixed
by passing a reference to an appropriate auto-class around, rather than
bare pointers.

runrevmark commented:
@livecodeali Done - I think the tests cover the various cases covered by the two bug reports.

livecodeali commented:
@livecode-vulcan review ok 8e404e3

livecode-vulcan commented:
💙 review by @livecodeali ok 8e404e3e9e17a886de5a64c024a6dd13736f2559

livecode-vulcan commented:
😎 test success 8e404e3e9e17a886de5a64c024a6dd13736f2559

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/452)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/444)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/444)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/444)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/444)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1966)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/194)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/193)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1997)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1987)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1986)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2014)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1955)
